src/translators/engines/google.ts

```diff
@@ -13,20 +14,18 @@ export default class GoogleTranslate extends TranslateEngine {
     } = options
 
     const key = Config.googleApiKey
-    const headers = key
-      ? { Authorization: `Bearer ${key}` }
-      : undefined
 
     const { data } = await axios({
       method: 'GET',
-      url: `${this.apiRoot}/translate_a/single?client=gtx&sl=${from}&tl=${to}&hl=zh-CN&dt=t&dt=bd&ie=UTF-8&oe=UTF-8&dj=1&source=icon&q=${encodeURI(options.text)}`,
-      headers,
+      url: key
+        ? `${this.apiRootIfUserSuppliedKey}/language/translate/v2?key=${key}&q=${encodeURI(options.text)}&source=${from}&target=${to}&alt=json`
```

Comment:
Can you lead me to its documentation? Sending the key in the query does not sound very safe to me.

Reply:
I agree that sending the key as a query parameter is suboptimal. These are the links I based my decision on:

https://cloud.google.com/docs/authentication/api-keys as well as some discussion in https://stackoverflow.com/questions/62148151/google-translate-api-authentication-key-and-usage

In the first link there's a part about API-key restrictions as well, so you can restrict it so that only your IP can use it, so in the eventual case where someone snipes your key, it'll be useless anyway.

If you can think of a better way of doing it, I'd be happy to change the PR.

If you want to go the route with using Auth headers on a request you need to generate new tokens with gcloud fairly often - which doesn't seem reasonable in the case of this addon.

Reply:
I will submit a minor change since the API protected endpoint doesn't support `auto` as a to/from parameter. It defaults to auto if you omit them, so will just not include the parameters if the option is missing/auto. Similar to how it looked before you added the default values.

Should be up shortly.

Reply:
Added a tiny new commit, tell me if you want me to squash them together.

Thanks for the speedy review btw :)